apis/v1/jaeger_webhook.go

```diff
@@ -12,10 +19,14 @@ const (
 )
 
 // log is for logging in this package.
-var jaegerlog = logf.Log.WithName("jaeger-resource")
+var (
+	jaegerlog = logf.Log.WithName("jaeger-resource")
+	cl        client.Client
```

Comment:
I don't like introducing a global state, but another option would be to implement the webhook "manually", which I like even less for validating and defaulting webhooks.

Reply:
Can we register another object to handle the webhook instead of Jaeger and embed Jaeger?

<details>
  <summary>Just an idea (untested)</summary>
  
```golang
type JaegerWebhook struct {
	Jaeger

	cl client.Client
}

type Jaeger struct {
	...
}
```
</details>

Reply:
that unfortunately does not work. We could add the client to the `Jaeger` but that is even worse.

Reply:
I think this is the "less worse" option we have, I don't like it either but I don't see any other possibility so far.

Reply:
While I was working on something else I became aware of the custom validator[0].

What do you think about using it? The method calls differ only slightly. In the `SetupWebhookWithManager` the locale client could be passed to the validator.

```golang
func (r *something) SetupWebhookWithManager(mgr ctrl.Manager) error {
	return ctrl.NewWebhookManagedBy(mgr).
		For(r).
		WithValidator(newSomethingValidator(mgr.GetClient())).
		Complete()
}
```

0: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/admission/validator_custom.go#L30-L35